src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -141,6 +141,16 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
         site_config.app_settings.append(NameValuePair(name="WEBSITE_NODE_DEFAULT_VERSION",
                                                       value=node_default_version))
 
+    # This is to keep the existing appsettings for a newly created webapp on existing webapp name.
+    name_validation = client.check_name_availability(name, 'Site')
+    if not name_validation.name_available:
+        existing_app_settings = _generic_site_operation(cmd.cli_ctx, resource_group_name,
+                                                        name, 'list_application_settings')
+        site_config.app_settings = []
+        site_config.app_settings += [NameValuePair(name=k, value=v)
```

Comment:
need to check updating existing app_settings: some config in app_settings already, and user want to update it.

Reply:
thanks will work and send the updated code